lib/workers/repository/finalise/repository-statistics.spec.ts

```diff
@@ -42,4 +52,86 @@ describe('workers/repository/finalise/repository-statistics', () => {
       );
     });
   });
+
+  describe('runBranchSummery', () => {
+    const cacheSpy = jest.spyOn(cache, 'getCache');
+
+    beforeEach(() => {
+      jest.resetAllMocks();
```

Comment:
why are u reseting all mocks?

Reply:
spy on mocks stay from 1 test to the other

Reply:
i resetting it because on every test the spy is being used to create a new mock, e,g, `cacheSpy.mockReturnValue`. so im making sure it is restored to its initial state before the next one. 

The spy indeed stays when in scope.

Reply:
i usually use mockReturnValueOnce, so it's only mocked once, and doesnt carry on to other tests

Reply:
im also using here the `Once` variant, but i guess i reseting it just to be sure. As even though it is `mockReturnValueOnce`, it might be restored to the value of `mockReturnValue` if previously set.

Reply:
don't reset just file sure, it hides wrong designed tests and causes pains when trying to debug a single test in future.

all tests should work standalone without interfere each other and without requiring to reset all in before each or some other function

Reply:
also creating a spy does nothing and still calls through until you add a mock implementation/ return value ...